packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -153,31 +151,38 @@ abstract class FlutterCommand extends Command<Null> {
     if (flutterUsage.isFirstRun)
       flutterUsage.printWelcome();
 
-    final FlutterCommandResult commandResult = await verifyThenRunCommand();
-
-    final DateTime endTime = clock.now();
-    printTrace("'flutter $name' took ${getElapsedAsMilliseconds(endTime.difference(startTime))}.");
-    if (usagePath != null) {
-      final List<String> labels = <String>[];
-      if (commandResult?.exitStatus != null)
-        labels.add(getEnumName(commandResult.exitStatus));
-      if (commandResult?.analyticsParameters?.isNotEmpty ?? false)
-        labels.addAll(commandResult.analyticsParameters);
-
-      final String label = labels
-          .where((String label) => !isBlank(label))
-          .join('-');
-      flutterUsage.sendTiming(
-        'flutter', 
-        name, 
-        // If the command provides its own end time, use it. Otherwise report
-        // the duration of the entire execution.
-        (commandResult?.endTimeOverride ?? endTime).difference(startTime), 
-        // Report in the form of `success-[parameter1-parameter2]`, all of which
-        // can be null if the command doesn't provide a FlutterCommandResult.
-        label: label == '' ? null : label,
-      );
+    FlutterCommandResult commandResult;
+    try {
+      commandResult = await verifyThenRunCommand();
+    } on ToolExit catch (_) {
+      commandResult = const FlutterCommandResult(ExitStatus.fail);
+      rethrow;
+    } finally {
+      final DateTime endTime = clock.now();
+      printTrace("'flutter $name' took ${getElapsedAsMilliseconds(endTime.difference(startTime))}.");
```

Comment:
flip the `"` and `'` over so the output uses `"` and the string itself uses `'`.